### ISSUE TITLE
compile with xcode issus (show wifi but not connect)

### ISSUE THREAD

**Issue by alighafourian** [OpenIntelWireless/itlwm#926]
**Have You Read Our Docs**
Yes 

**Are You Reporting A Bug**
Yes
**Environment**

- Kext Version: 
- WiFi Card Model:7265N 
- PCI Product ID: (0x8086, 0x5002)
- macOS Version: 14.1

**Description**
I download code file and compiler . compile done but show wifi and dosnt connect 

**Bug Report Archive**

![Screenshot 2023-10-24 at 12 50 05](https://github.com/OpenIntelWireless/itlwm/assets/51196916/456901a9-a6b1-4293-ab78-c4ecfba706fe)
![Screenshot 2023-10-24 at 12 52 24](https://github.com/OpenIntelWireless/itlwm/assets/51196916/e7b62d8c-08f6-48b1-899f-53c1d7b2b3b4)
![Screenshot 2023-10-24 at 12 54 37](https://github.com/OpenIntelWireless/itlwm/assets/51196916/275cf0bf-9327-43b9-a04c-34ff9b7f178f)

kext log:
[Log_2023-10-24_12-41-37.log](https://github.com/OpenIntelWireless/itlwm/files/13112983/Log_2023-10-24_12-41-37.log)

**Kext Download Source**

**Comment by ShroomKing** [comment]
Same when building from source with latest Xcode, release build works fine though.

**Comment by alighafourian** [comment]
@ShroomKing 
only way fork on github and run xcode Workflow

**Comment by alighafourian** [comment]
ok i close issus

**Comment by alighafourian** [comment]
@diepeterpan 
ok i close issus

**Comment by ShroomKing** [comment]
I did more tests.
This appears to be a compiler code optimisation problem.

When creating a debug build it uses optimisation level 0 and the kext works.

But when creating an optimised build (Release build type / Product->Archive) the kext can't connect to wifi.

The highest optimisation level that works right now is 1, so I assume some flag in level 2 is breaking the code (more info here: https://gcc.gnu.org/onlinedocs/gcc/Optimize-Options.html)

You can disable optimisation flags by using `-fno-(flag-name)` so for example to disable `-falign-functions` add `-fno-align-functions` to CFLAGS, this way we can keep the other optimisations but I don't know if there's a better way to debug this than trying to disable all the flags one by one.

**Comment by usr-sse2** [comment]
> The highest optimisation level that works right now is 1, so I assume some flag in level 2 is breaking the code (more info here: https://gcc.gnu.org/onlinedocs/gcc/Optimize-Options.html)

It's better to look options in Clang documentation because Xcode compiles with Clang, not GCC.

**Comment by zxystd** [comment]
> > The highest optimisation level that works right now is 1, so I assume some flag in level 2 is breaking the code (more info here: https://gcc.gnu.org/onlinedocs/gcc/Optimize-Options.html)
> 
> It's better to look options in Clang documentation because Xcode compiles with Clang, not GCC.

@usr-sse2 You are right, but I don't have the time and environment(no Sonoma/new XCode) to investigate this, if you have discoveries please let me know or open an PR, thank you!

**Comment by zxystd** [comment]
From the log I can see the root issue is that the optimization breaks function `ieee80211_get_assoc_req`, with invalid assoc request(RSN invalid), for what's the differences I can't see with the decompiled code.

**Comment by vanyasem** [comment]
@zxystd I can confirm that the issue is resolved with the latest commits. Release builds are now able to connect to Wi-Fi successfully. Tested on Sonoma 14.4

**Comment by alighafourian** [comment]
> @zxystd I can confirm that the issue is resolved with the latest commits. Release builds are now able to connect to Wi-Fi successfully. Tested on Sonoma 14.4

Compile with xcode app 
If use terminal for build wifi can not connect  but show 
If use xcode app work good
